modules/streamer/src/main/BsonHandlers.scala

```diff
@@ -6,16 +6,7 @@ import reactivemongo.api.bson.*
 
 private object BsonHandlers:
 
-  given BSONDocumentHandler[Streamer.YouTube] = new BSON[Streamer.YouTube]:
-    def reads(r: BSON.Reader) =
-      Streamer.YouTube(
-        r.get[String]("channelId"),
-        r.getO[String]("liveVideoId"),
-        r.getO[String]("pubsubVideoId")
-      )
-    def writes(w: BSON.Writer, o: Streamer.YouTube) =
-      $doc("channelId" -> o.channelId, "liveVideoId" -> o.liveVideoId, "pubsubVideoId" -> o.pubsubVideoId)
-
+  given BSONDocumentHandler[Streamer.YouTube]  = Macros.handler
```

Comment:
Defining this handler solved YouTube objects failing to load under the macro. Maybe because of the options within an option?